SAUCE_LABS_HOSTED_APPIUM.md

```diff
@@ -0,0 +1,141 @@
+
+## The Test-per-Session Model vs. The Suite-per-Session Model
+
+Think about a traditional Appium test run. For every single test, you typically have to:
+
+**Traditional Model (Test-per-Session):**
+1. Wait for a device to become available.
+2. Wait for the device to be prepared.
+3. Wait for the Appium session to start.
+4. Run your test.
+5. Tear down the session and release the device.
+6. Repeat for every test in your suite.
+
+This cycle means a lot of your pipeline's time is spent on repetitive setup and teardown, not on executing tests.
+
+**The OpenAPI Session Model (Suite-per-Session):**
+
+***The OpenAPI session model transforms this workflow.*** Instead of reserving a device for each individual Appium test, you reserve a 
+device once for the entire suite. You can then run any number of tests on that single, persistent device session.
+
+1. Start a device session
+2. Start Appium server once.
+3. Run Test 1.
+4. Run Test 2.
+5. ...Run Test N
+6. Tear down the session once.
+
+
+## Core Value for Testers: Speed, Control, and Efficiency
+
+### Blazing Speed: Go from Minutes to Seconds
+By paying the "startup cost" of getting a device and launching Appium only once, your subsequent tests become faster.
+
+* ***Eliminate Redundant Waits:*** No more waiting for a new device for each test. The device is yours for the duration of
+  the test suite.
+
+* ***Instant Test Execution:*** Your second, third, and fourth tests can start immediately after the previous one finishes.
+  This drastically shortens the feedback loop, allowing you to find—and verify fixes for—bugs faster than ever before.
+
+### Unprecedented Control: Become a Device Orchestrator
+An open session transforms the remote device into your personal workbench. You are no longer just running a test;
+you are orchestrating the device's state to create more powerful and realistic test scenarios.
+
+* ***Run Dependent Tests:*** Easily test workflows that span multiple steps, like adding an item to a cart in one test and
+  checking out in another, all without restarting the app.
+
+* ***Manipulate Device State:*** Between tests, you have full control. You can programmatically:
+  * Push test data, files, or pre-configured settings.
+  * Clear app cache or user data to start the next test from a clean slate.
+  * Run diagnostic scripts.
+
+* ***Debug with Precision:*** If a test fails, the device and app remain in their failed state. You can keep the session
+  open, get the Live View URL, and manually inspect the device to understand exactly what went wrong.
+
+### Improved Code Structure
+
+This approach promotes a cleaner separation of concerns. As the example below shows, the complex logic 
+for reserving a device and starting Appium is centralized into a one-time suite setup. This leaves your individual `@Test` methods leaner 
+and more focused on their specific actions and assertions, making them easier to read and maintain.
```

Comment:
This is in relation to your specific setup. 

I don't get why using open api automatically leads to improved code structure. 

Using just appium capabilities, as people do currently, is very clean. They don't need to manage all the device config. 

I don't think we can make this broad claim here. `This leaves your individual `@Test` methods leaner` is a big assumption